index/scorch/persister.go

```diff
@@ -435,19 +438,38 @@ func (s *Scorch) removeOldData() {
 // NumSnapshotsToKeep represents how many recent, old snapshots to
 // keep around per Scorch instance.  Useful for apps that require
 // rollback'ability.
-var NumSnapshotsToKeep int
+var NumSnapshotsToKeep = 1
 
 // Removes enough snapshots from the rootBolt so that the
 // s.eligibleForRemoval stays under the NumSnapshotsToKeep policy.
 func (s *Scorch) removeOldBoltSnapshots() (numRemoved int, err error) {
-	var epochsToRemove []uint64
+	persistedEpochs, err := s.rootBoltSnapshotEpochs()
+	if err != nil {
+		return 0, err
+	}
+
+	if len(persistedEpochs) <= NumSnapshotsToKeep {
+		// we need to keep everything
+		return 0, nil
+	}
+	protectedEpochs := make(map[uint64]struct{}, NumSnapshotsToKeep)
+	// now make this a list of epochs to protect from deletion
```

Comment:
minor comment thing -- making a map here instead of a list?

Reply:
oops ya thats old comment i missed, ill clean up once i get all the feedback